apis/v1alpha1/backendpolicy_types.go

```diff
@@ -55,33 +64,43 @@ type BackendPolicySpec struct {
 	// the oldest BackendPolicy.
 	//
 	// Support: Core
+	//
+	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []BackendRef `json:"backendRefs"`
 
 	// TLS is the TLS configuration for these backends.
 	//
 	// Support: Extended
+	//
 	// +optional
-	TLS *BackendTLSConfig `json:"tls,omitempty"`
+	TLS BackendTLSConfig `json:"tls,omitempty"`
```

Comment:
Although the API conventions do have an allowance for types with a built-in nil value, it seems like the overwhelming preference is to use pointers for all optional fields if I'm reading the API conventions correctly.

Reply:
> preference is to use pointers for all optional fields

Not all optional fields.

> They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) __or have a built-in nil value (e.g. maps and slices).__

We are inconsistently using pointers for struct fields.

Reply:
Yeah, the conventions are a bit confusing here. I completely agree that they allow for non-pointer optional fields in these scenarios, but the points below seem to suggest we should be using pointers for structs:

> * structs are not omitted from encoder output even where omitempty is specified, which is messy;
> * having a pointer consistently imply optional is clearer for users of the Go language client, and any other clients that use corresponding types

Reply:
From an implementer's perspective, checking for a nil pointer to a struct is far less error prone than checking each member in that struct.

Reply:
@roberthbailey @hbagdi thank you for the feedback. If pointers are used for optional structs, I would like to be consistent across the API's. This means changing spec/status for all resources. For example:
```
Spec GatewaySpec `json:"spec,omitempty"`

// change to:
Spec *GatewaySpec `json:"spec,omitempty"`
```

By doing so, spec/status optionality will differ from most [k8s APIs](https://github.com/kubernetes/api).

Reply:
I haven't seen any k8s resources using pointers for meta, spec, or status. I'm not sure why that is, I really don't even understand why they're optional, but I'd rather not diverge from the pattern in upstream for these fields.